src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -890,7 +889,8 @@ export class Kernel implements IKernel {
             );
             results.push(...configInit.splitLines({ trim: false }));
         }
-        return results;
+        // Wrap in try..except in case user doesn't have matplot lib installed.
+        return ['try:', ...results.map((line) => `    ${line}`), 'except:', '   pass'];
```

Comment:
The matplot lib code is the only thing that can fail, added try..except around that, 
The rest should never fail, if it does we want things to fail rather than silently failing
I.e. we need to avoid unnecessarily swallowing all exceptions

Reply:
> The matplot lib code is the only thing that can fail, added try..except around that

There has been other failures in the past. Like the code that sets the '__vscipnb_file_name__'. That failed because of invalid characters in the substitution. 

The same can be said for the startup code. 

And any failure that happens in a executeSilent is never shown to the user. I think we should try/except each one.

Reply:
Or maybe show them to the user if the fail. That's probably better for the startup commands at least. Maybe that one should be special cased because the user entered it.

Reply:
Sorry I thought that I was working on this. I have it added on my PR.

Reply:
This is what I was adding. Missed Don's initial PR coming in for some reason. https://github.com/microsoft/vscode-jupyter/pull/9197/commits/a19e1b9682fcfe4bba2a7cba9fc1d6b02b5f2dfb

Reply:
I'm logging the errors instead of displaying them,
If you have startup code today, it will apply to all kernels. anyone who has startup code for Python and runs a non-python kernle will all of a sudden get failure error messages displayed.

We have a seprate issue to have startup code per language, we'd need to address that first, else we coudl end up spamming the user unnecessarily.

Reply:
> Sorry I thought that I was working on this. I have it added on my PR.

Sorry, but didn't see the issue has being assigned to anyone hence I started on this.